includes/admin/class-senseilms-plugin-updater.php

```diff
@@ -0,0 +1,232 @@
+			$instance = new self( SENSEI_IB_PLUGIN_FILE, SENSEI_IB_VERSION );
+		} elseif ( class_exists( '\Sensei_Pro\Setup_Context' ) ) {
+			$instance = new self( SENSEI_PRO_PLUGIN_FILE, SENSEI_PRO_VERSION );
+		} else {
+			return;
+		}
+
+		add_filter( 'plugins_api', [ $instance, 'get_plugin_info' ], 15, 3 );
+		add_filter( 'site_transient_update_plugins', [ $instance, 'maybe_inject_custom_update_to_update_plugins_transient' ], 15 );
+	}
+
+	/**
+	 * Get plugin information as expected by the `plugins_api` hook.
+	 * This will be called to display the details for the updated in the detailed view.
+	 *
+	 * @param false|object|array $res    Result. As defined per the `plugins_api` hook.
+	 * @param string             $action The action being executed. As defined per the `plugins_api` hook.
+	 * @param object             $args   The arguments. As defined per the `plugins_api` hook.
+	 *
+	 * @hooked plugins_api
+	 *
+	 * @return false|object If other than false is returned the actual call to wordpress.org is not done.
+	 */
+	public function get_plugin_info( $res, $action, $args ) {
+		if (
+			'plugin_information' !== $action
+			|| $this->plugin_slug !== $args->slug
+			|| false !== $res
+		) {
+			return $res;
+		}
+
+		$remote = $this->request_info();
+		if ( is_wp_error( $remote ) ) {
+			// Early return in case request to SenseiLMS.com failed.
+			return $res;
+		}
+
+		$res                = new stdClass();
+		$res->name          = $remote->name;
+		$res->slug          = $remote->slug;
+		$res->author        = $remote->author;
+		$res->version       = $remote->version;
+		$res->requires      = $remote->requires;
+		$res->tested        = $remote->tested;
+		$res->requires_php  = $remote->requires_php;
+		$res->last_updated  = $remote->last_updated;
+		$res->sections      = [
+			'description'  => $remote->sections->description,
+			'installation' => $remote->sections->installation,
+			'changelog'    => $remote->sections->changelog,
+		];
+		$res->download_link = $remote->download_url;
+		$res->banners       = [
+			'low'  => $remote->banners->{'1x'},
+			'high' => $remote->banners->{'2x'},
+		];
+
+		// TODO: ADD CSS.
+		wp_enqueue_style( 'sensei-updater-styles', plugin_dir_url( $this->plugin_full_name ) . 'assets/dist/senseilms-licensing/styles.css', [], $this->version );
+
+		return $res;
+	}
+
+	/**
+	 * Potentially injects the details for a new plugin version by checking against the remote server.
+	 * This is done by hooking into the `update_plugins` transient by using the `site_transient_update_plugins` hook.
+	 *
+	 * @param mixed $transient The plugin_update transient.
+	 *
+	 * @hooked site_transient_update_plugins See reference for `site_transient_transient`.
+	 *
+	 * @return mixed
+	 */
+	public function maybe_inject_custom_update_to_update_plugins_transient( $transient ) {
+
+		// Skip empty transients or if it was already set by Sensei Pro.
+		if ( empty( $transient ) || isset( $response[ $this->plugin_full_name ] ) ) {
```

Comment:
If `$response[ $this->plugin_full_name ]` is already set, it means that Sensei Pro already handled it. So we don't need to do it.

Reply:
While testing this I saw an error: it should be `$transient[ $this->plugin_full_name ]` and not `$response`